### DISCUSSION TITLE
Woche 03 (Sem4) - Clean Code

### DISCUSSION THREAD

**Discussion by niklaskrauth** [mausio/corgi-shop-doc#13] [Blogs]
In diesem Blogeintrag wird gezeigt wie wir Clean Code Prinzipien genutzt haben um unser Projekt zu refactorn sowie auch Updates seit dem letzten Post.
## Clean Code

Im Frontend als auch im Backend werden weitverbreitete Clean Code Prinzipien wie DRY oder KISS schon während der Produktion direkt angewandt. Trotzdem wurde im letzten Refactor ungenutzter Code gelöscht, welcher es uns vereinfacht hatte Testdaten in die lokale Datenbank zu speichern. Dadurch das aber nun eine Datenbank auf einem statischen Server läuft sind die Testdaten irrelevant.

## Updates

Dazu wurde ein User-Objekt im Backend erstellt welches Name, Email und Passwort besitzt. Wenn man sich im Frontend anmeldet werden die Daten an das Backend weitergeleitet und ein JWT-Token wird wieder zurück gesendet. Darüber kann das Frontend mit dem Backend weiterhin kommunizieren während der User angemeldet ist.

**Comment by Unkn0wncod3** [comment]
Tolle Arbeit beim Umsetzen der Clean Code Prinzipien. Die Einführung des User-Objekts im Backend und die Verwendung von JWT-Token für die Authentifizierung sind klare Beispiele dafür, wie ihr das KISS-Prinzip in eurem Projekt umsetzt.

**Comment by ghost** [comment]
Kling sehr spannend. Einen großen Forschritt habt ihr anscheinend im Bereich der Datenbank, dass ihr nun keine Testdaten auf einer lokalen Datenbank habt, sondern nun eine remote DB verwendet. Das ihr DRY und KISS verwendet klingt ebenfalls gut, jedoch interessiert mich noch ob ihr auch auf andere bekannte Prinzipien wie SOLID geachtet habt. Ebenfalls interessiert mich, ob ihr in eurem Swift Frontend auch Swift/Mobile App spezifischen Clean Code/Design Patterns verwendet?
